nilearn/surface/surface.py

```diff
@@ -1014,3 +1014,235 @@ def check_surface(surface):
     mesh, data = check_mesh_and_data(surface.mesh,
                                      surface.data)
     return Surface(mesh, data)
+
+def compute_adjacency_matrix(surface, values='ones', dtype=None):
```

Comment:
One question about the logic between private and public functions (excuse my ignorance since Python is not my native language): I could see this function being useful for many other things. For example doing clustering on the surface and doing various cluster based thresholding and permutations. In that case, is it better to be public or private? I assume private is harder for users to find so, therefor, someone who would be inclined to write surface based clustering might not realize that most of the heavy lifting is already done here

Reply:
On the contrary to other programming languages, there is no "real" difference between pubic and private functions in Python. Both can be imported and used in the same way.
The key difference is that public functions are part of the project's API that we expose to the users. These functions and classes are properly documented and assumed to be stable and backward compatible. This means that, once an object becomes public it is difficult (and long because of deprecation cycles) to change it, since we are bound to a backward compatibility contract with the users of the library. On the contrary, we are free to change private objects "as we want", which is much easier in terms of maintenance.
I had the feeling that these two functions were helper functions for `smooth_surface_data`, but you are right that they could be directly useful for users, and therefore be public and part of the Nilearn API.
If we agree that this is the case, we should think about the best location for them as well as potential exemples to showcase them. Ping @alexisthual @bthirion @ymzayek

Reply:
In our case, I think these functions should be kept private. They could indeed be useful to some users (I myself have a personal repo for this kind of util functions only haha) but then a whole bunch of similar util functions could also be integrated. I'm afraid this would make the API look fuzzy on the long run